src/ShortcutConfigurator.cc

```diff
@@ -0,0 +1,454 @@
+                                return;
+                            }
+                            else if(shortcut!=QString::fromUtf8("")) 
+                            {
+                                shortcutOccupied.insert(shortcut,action);
+                            }
+                            shortcutsListFinal.append(shortcut);
+                        }
+                    }
+
+                    action->setShortcuts(shortcutsListFinal);
+                }   
+            }
+        }
+    }
+
+}
+
+QString ShortcutConfigurator::getData(int row,int col)
+{
+    return shortcutsTable->model()->index(row,col).data().toString();
+}
+
+void ShortcutConfigurator::putData(QModelIndex indexA,QString data)
+{
+    shortcutsTable->model()->setData(indexA, data);
+}
+
+void ShortcutConfigurator::raiseError(const QString errorMsg)
+{
+        QMessageBox messageBox;
+        messageBox.critical(0,"Error",errorMsg);
+        messageBox.setFixedSize(500,200);
+}
+
+void ShortcutConfigurator::onTableCellClicked(const QModelIndex & index)
+{
+    if (index.isValid() && index.column()!=0) 
+    {
+        //just to avoid opening of multiple dialogs on multiple clicks on cell
+        if(shortcutCatcher) delete shortcutCatcher;
+
+        shortcutCatcher = new QMessageBox;
+        QString updatedAction = getData(index.row(),0);
+        auto itr = shortcutsMap.find(updatedAction);
+        QAction* changedAction = itr.value();
+
+        shortcutCatcher->raise();
+        shortcutCatcher->setStyleSheet("QLabel{min-width: 400px;}");
+        shortcutCatcher->setInformativeText(QString(""));
+        shortcutCatcher->setWindowTitle(QString("Set Shortcut for: ")+updatedAction);
+        shortcutCatcher->installEventFilter(this);
+        shortcutCatcher->setText("Press the Key Sequence");
+        shortcutCatcher->setStandardButtons(QMessageBox::Apply | QMessageBox::Reset | QMessageBox::Cancel);
+        shortcutCatcher->setDefaultButton(QMessageBox::Apply);
+        shortcutCatcher->setWindowModality(Qt::WindowModal);
+        int ret = shortcutCatcher->exec();
+        QString updatedShortcut;
+
+        switch (ret)
+        {
+            case QMessageBox::Apply:
+                updatedShortcut = pressedKeySequence.toString(QKeySequence::NativeText);
+                shortcutCatcher->close();
+                break;
+            case QMessageBox::Reset:
+                updatedShortcut = QString("");
+                shortcutCatcher->close();
+                break;
+            case QMessageBox::Cancel:
+                shortcutCatcher->close();
+                return;
+        }
+
+        if(shortcutOccupied[updatedShortcut])
+        {
+            raiseError(QString("Shortcut Already Occupied By: "+shortcutOccupied[updatedShortcut]->objectName()));
+            return;
+        }
+
+        QList<QKeySequence> shortcutsListFinal;
+
+        bool singleShortcutUpdate = false;
+
+        if(index.column()==1)
```

Comment:
Thanks, I should have looked at it one more time, as it could have been optimized.
I think, leaving the primary empty will be better than shifting, because of that empty string in JSON, to disable the shortcut.